packages/astro/src/content/loaders/file.ts

```diff
@@ -1,25 +1,54 @@
 import { promises as fs, existsSync } from 'node:fs';
 import { fileURLToPath } from 'node:url';
+import yaml from 'js-yaml';
 import { posixRelative } from '../utils.js';
 import type { Loader, LoaderContext } from './types.js';
 
+export interface FileOptions {
+	/**
+	 * the parsing function to use for this data
+	 * @default JSON.parse or yaml.load, depending on the extension of the file
+	 * */
+	parser?: (text: string) => any;
```

Comment:
I went with unknown. I don't think generics are very helpful here because most parse functions (JSON.parse, yaml.load, "toml".parse) return `unknown` or `any`